frontend/src/screens/study-landing.tsx

```diff
@@ -18,20 +18,59 @@ import {
     Text,
     Title,
 } from '@mantine/core';
+import { IconCheck } from '@tabler/icons-react'
 import Markdown from 'react-markdown'
 import { useLearningPathStudies } from './learner/studies';
 import { CompactStudyCard } from '../components/study/compact-study-card';
+import { notifications } from '@mantine/notifications';
+
+let notificationShownThisSession = false;
 
 export default function StudyLanding() {
     const env = useEnvironment()
-
     const study = useLoaderData() as ParticipantStudy
     const learningPathStudies = useLearningPathStudies(study?.learningPath)
 
+    const showEarnedPointsNotification = (points: number) => {
+        if (notificationShownThisSession) return;
+        
+        notifications.show({
+            title: `You just earned ${points} points!`,
+            message: 'The longer the study, the more points you earn. Reach 200 points to unlock additional rewards.',
+            icon: <IconCheck size="1.1rem" />,
+            color: 'teal',
+            autoClose: 5000, 
+            styles: () => ({
+                description: { fontSize: '12px' },
+            }),
+        });
+        
+        notificationShownThisSession = true;
+    };
+
+    useEffect(() => {
+        if (study.totalPoints > 0) {
+            // Delay to ensure it runs after any potential double renders
+            const timer = setTimeout(() => {
+                showEarnedPointsNotification(study.totalPoints);
+            }, 100);
+            
+            return () => clearTimeout(timer);
+        }
+    }, [study.totalPoints]);
```

Comment:
what happens if you remove `study.totalPoints` from the dependency array? ideally this should just run once when the component renders initially right?

Reply:
yes but, when i did that for some weird reason, the notifcations got displayed twice, one above another like this: 
![image](https://github.com/user-attachments/assets/74d90900-d8c3-4b7a-a538-53c46c7ea015)

Reply:
this could be just a side effect of [react rendering components twice in dev mode](https://chanduthedev.medium.com/why-react-app-rendering-twice-e31339ed9461#:~:text=StrictMode%20in%20the%20application%2C%20react,environment%2C%20it%20only%20renders%20once.&text=If%20you%20want%20to%20avoid,the%20development%20environment%2C%20remove%20React.)